orc/src/main/java/org/apache/iceberg/orc/OrcFileAppender.java

```diff
@@ -153,4 +182,10 @@ private static Writer newOrcWriter(OutputFile file,
                                                          createWriterFunc) {
     return (OrcRowWriter<D>) createWriterFunc.apply(schema, orcSchema);
   }
+
+  private TreeWriter treeWriterHiddenInORC() {
+    DynFields.BoundField<TreeWriter> treeWriterFiled =
+        DynFields.builder().hiddenImpl(writer.getClass(), "treeWriter").build(writer);
```

Comment:
Pls add a TODO saying that in which ORC version we will turn to access the `estimateMemorySize` directly without accessing this hidden `treeWriter` to get this value ?